linux_files/setup

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+set -e
+
+# functions
+
+## core functions
+
+function welcomeprompt {
+echo "Welcome to WLinux."
+echo 
+echo "WLinux comes with a core set of packages. This script allows you to install optional sets of packages."
+echo
+}
+
+function continueprompt {
+read -r -p "Would you like to continue setup? [Y/n] " response
+if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]
+then
+    echo "Continuing with setup."
+else
+    echo "You may run the script again any time by running: bash /etc/setup"
+    quit
+fi
+}
+
+function quit {
+exit
+}
+
+function updatescriptprompt {
+read -r -p "Would you like to download and run an updated version of this setup from GitHub? [y/N] " response
+if [[ $response == ^[yY]$ ]]; then
+then
+    TMPDIR=$(mktemp -d)
+    curl https://raw.githubusercontent.com/WhitewaterFoundry/WLinux/sirredbeard/linux_files/setup | | bash -s -- u
+    quit
+else
+    echo "Continuing with release install script."
+fi
+}
+
+function securitypatches {
+}
+
+## package meta functions
+
+function installpackages {
+zshinstall
```

Comment:
I think for this part, isn't better to select what shell to use instead of asking one by one? like `select your preferred shell `, also for editor, but I think it is good enough for now.

Reply:
I want that but we need to populate the various shell and editor options and set up the selection prompt/handling. What if they want multiple?

Reply:
hmm, yeah, but then maybe separate chsh part?